src/voiceClient.js

```diff
@@ -0,0 +1,347 @@
+     * 
+     * @param {string} query 
+     * @returns 
+     */
+    async loadSource(query) {
+        const { source } = await this.http.loadSource(query)
+
+        return source
+    }
+
+    /**
+     * 
+     * @param {number} [offset=1]
+     * @returns 
+     */
+    async skip(offset = 1) {
+        return await this.http.skip(offset)
+    }
+
+    /**
+     * 
+     * @param {number} offset
+     * @returns 
+     */
+    async seek(offset) {
+        return await this.http.seek(offset)
+    }
+
+    async getOptions() {
+        return await this.http.getOptions()
+    }
+
+    /**
+     * 
+     * @param {DiscodoAudioOptions} options 
+     * @returns 
+     */
+    async setOptions(options) {
+        if ("volume" in options) this._volume = options.volume
+        if ("crossfade" in options) this._crossfade = options.crossfade
+        if ("autoplay" in options) this._autoplay = options.autoplay
+        if ("filter" in options) this._filter = options.filter
+
+        return await this.http.setOptions(options)
+    }
+
+    async setVolume(volume) {
+        return await this.setOptions({ volume })
+    }
+
+    async setCrossfade(crossfade) {
+        return await this.setOptions({ crossfade })
+    }
+
+    async setAutoplay(autoplay) {
+        return await this.setOptions({ autoplay })
+    }
+
+    async setFilter(filter) {
+        return await this.setOptions({ filter })
+    }
+
+    async pause() {
+        return await this.http.pause()
+    }
+
+    async resume() {
+        return await this.http.resume()
+    }
+
+    async shuffle() {
+        const data = await this.http.shuffle()
+
+        this.queue.handleGetQueue(data)
+
+        return this.queue
+    }
+
+    async getCurrent() {
+        return this._current = await this.http.getCurrent()
+    }
+
+    async fetchState() {
+        return await this.query("getState")
+    }
+
+    /**
+     * 
+     * @param {boolean} [ws=true]
+     * @returns 
+     */
+    async fetchQueue(ws = true) {
+        if (ws) await this.query("getQueue")
+        else this.queue.handleGetQueue(await this.http.queue())
+
+        return this.queue
+    }
+
+    async requestSubtitle(lang, url) {
+        if (!lang && !url) throw new Error("Either `lang` or `url` is needed.")
+
+        const Data = {}
+
+        if (url) {
+            Data.url = url
+        } else if (lang) {
+            Data.lang = lang
+        }
+
+        return await this.query("requestSubtitle", Data)
+    }
+
+    async getSubtitle() {
+        throw new Error("not implemented")
+    }
+
+    async moveTo(node) {
+        if (node === this.node) throw new Error("Already connected to this node.")
+
+        const channel = this.client.client.channels.cache.get(this.channelID)
+        if (!channel) throw new Error("this voice client is not connected to the channel.")
+
+        const [VC] = await Promise.all([
+            this.client.connect(channel, node), 
+            this.fetchState()
+        ])
```

Comment:
Should fetch the current state before connect the other node

```suggestion
        await this.fetchState()
        
        const [VC] = await this.client.connect(channel, node)
```